examples/00-released_examples/06-lunar_lander_thermal.py

```diff
@@ -0,0 +1,858 @@
+# It returns the Sun's altitude and azimuth as outputs.
+#
+# For more information, see the following links:
+#
+# * `Ecliptic longitude <https://w.wiki/7zFR>`_
+# * `Subsolar longitude <https://w.wiki/7zFS>`_
+# * `Altitude and azimuth <https://w.wiki/7zFU>`_
+
+
+def calc_sun_vecs_for_moon(
+    earth_ecliptic_lon_start: float,
+    subsol_lon_start: float,
+    obsv_lat: float,
+    obsv_lon: float,
+    t: float,
+) -> tuple[float, float]:
+    # Earth ecliptic longitude
+    earth_ecliptic_lon = earth_ecliptic_lon_start + 2 * np.pi / (365 * 86400) * t
+
+    # Subsolar point
+    subsol_lat = np.arcsin(np.sin(-moon_obliquity) * np.sin(earth_ecliptic_lon))
+    subsol_lon = subsol_lon_start + t / (29.5 * 86400) * 2 * np.pi
+
+    # Solar altitude and azimuth
+    sun_obsv_phaseang = np.arccos(
+        np.sin(subsol_lat) * np.sin(obsv_lat)
+        + np.cos(subsol_lat) * np.cos(obsv_lat) * np.cos(obsv_lon - subsol_lon)
+    )
+
+    sun_alt = np.pi / 2 - sun_obsv_phaseang
+    sun_azm = np.arctan2(
+        np.cos(obsv_lat) * np.sin(subsol_lat)
+        - np.sin(obsv_lat) * np.cos(subsol_lat) * np.cos(subsol_lon - obsv_lon),
+        np.cos(subsol_lat) * np.sin(subsol_lon - obsv_lon),
+    )
+
+    return sun_alt, sun_azm
+
+
+###############################################################################
+# Define beam direction function
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Define the function that converts the Sun's altitude and azimuth, as
+# calculated by ``calc_sun_vecs_for_moon``, into a beam direction in Cartesian
+# coordinates that can be used in Fluent. The coordinate system used is:
+#
+# * X: North
+# * Y: Zenith
+# * Z: East
+
+
+def sun_vec_to_beam_dir(
+    sun_alt: float,
+    sun_azm: float,
+) -> tuple[float, float, float]:
+    # Coordinate system:
+    #   X: North
+    #   Y: Zenith
+    #   Z: East
+    x = np.cos(sun_alt) * np.cos(sun_azm)
+    y = np.sin(sun_alt)
+    z = np.cos(sun_alt) * np.sin(sun_azm)
+
+    # Since the sun vector points toward the sun, we need to take the
+    # negative to get the beam direction
+    return -x, -y, -z
+
+
+###############################################################################
+# Define mean surface temperature function
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# The radiator will be opened or closed based on its mean temperature. We will
+# access the solver session object to extract nodal temperatures. It takes a
+# list of surface names as input, finds their surface IDs, obtains the scalar
+# field data from the solver, then returns the average temperature.
+
+
+def get_surf_mean_temp(
+    surf_names: list[str],
+    solver: pyfluent.session_solver.Solver,
+) -> float:
+    # Get surface IDs
+    surfs = solver.field_info.get_surfaces_info()
+    surf_ids_ = [surfs[surf_name]["surface_id"] for surf_name in surf_names]
+
+    # Flatten surf_ids nested list
+    surf_ids = []
+    _ = [surf_ids.extend(sublist) for sublist in surf_ids_]
```

Comment:
```suggestion
    surf_ids = list(chain(*surf_ids_))
```
(requires `from itertools import chain`)